### DISCUSSION TITLE
The vanishing cargo, or how to load 5000 t in a 50 t cargo hold?

### DISCUSSION THREAD

**Discussion by esksp** [endless-sky/endless-sky#10717] [General]
I have several ships, with varying cargo space sizes: My flagship, able to hold 100 t, a freighter able to hold 900 t, and a second warship able to hold 100 t.

Now imagine I disable a Korath raider ship, and collect a couple useful items sticking out.
I can only steal up to 100 t of stuff, because that's how much my flagship can carry, and there is no easy way to transfer items to the other ships flying nearby. So far so good, that's all very logical.

Now I fly back and land on a friendly planet, and would like to move those (nearly) 100 t of cargo from my 100 t flagship to the 900 t freighter. Oops, no can do: That cargo is still there, I can see (and sell) it in the Outfitter, but all my ships now say they're empty! Where did my cargo go?
Can I just take off and go collect another 100 t worth of Korath equipment, or will that phantom cargo appear again once I'm in space? And _where_ exactly will it appear?...
Or can I keep collecting cargo, just land to make it disappear in some parallel dimension, and then immediately take off to go collect some more?

I think there is some room for improvement there, unless of course I've missed something obvious, in which case I'd be grateful if somebody would care to explain... :-D

**Comment by TomGoodIdea** [comment]
When you land on a planet, your cargo is pooled to a single "cargo hold" that has the capacity of all your ships' combined cargo space. When you depart, any cargo you have _should_ be redistributed in your fleet (with priority to give you as much free space on the flagship as possible).

**Reply by esksp** [reply]
Thanks.
Would it be possible to allow the player to choose which ship will get that cargo?
Cargo space is vital for many things, and some cargo management would help, for instance allowing you to pool all your carry-around cargo into one or more _dedicated_ cargo ships, leaving your warships free for collecting interesting items you might stumble upon, or simply avoiding to make them sluggish due to the increased mass.
I understand this would probably require a substantial rewrite. A compromise solution would be to preferentially load cargo into any "Freighter" type ships you might have, and only put some into your warships if there is not enough free Freighter space (with your flagship being the last to get some). Just an idea.
